File: homework/src/h10/h10.rb
Pair = Struct.new(:fst,:snd)

def summingPairs(xs, sum)
  the_pairs = []
  len = xs.length

  for i in 0..(len-1)
    for j in (i+1)..(len-1)
      if xs[i] + xs[j] <= sum
        the_pairs.push(Pair.new(xs[i],xs[j]))
      end
    end
  end

  return the_pairs
end

require 'date'
require_relative 'collection'

puts "Starting at:   #{DateTime.now.sec} seconds, #{DateTime.now.strftime("%9N")} nanoseconds"
puts summingPairs(INPUT,2020)
puts "Ending at:     #{DateTime.now.sec} seconds, #{DateTime.now.strftime("%9N")} nanoseconds"

File: homework/src/h10/h10.clj
(defn summingPairs [xs sum]
  (letfn [(summingPairsHelper [xs the_pairs]

            (if (empty? xs) the_pairs


                (let [[fst & rest] xs]


                  (recur
                   rest






                   (doall 
                    (concat the_pairs
                            (for [snd rest ;; For each `snd` in `rest`...
                                  :when (<= (+ fst snd) sum)]

                              [fst snd])))))))]
    (summingPairsHelper xs [])))

(load-file "./collection.clj")

(println (str
          "Starting at:   "
          (.getSecond (java.time.LocalDateTime/now))
          " seconds, "
          (.getNano (java.time.LocalDateTime/now))
          " nanoseconds"))
(println (summingPairs input 2020))
(println (str
          "Ending at:     "
          (.getSecond (java.time.LocalDateTime/now))
          " seconds, "
          (.getNano (java.time.LocalDateTime/now))
          " nanoseconds"))
